src/client/index.tsx

```diff
@@ -24,59 +24,80 @@ const notebookApi = acquireNotebookRendererApi(JupyterNotebookRenderer);
 
 notebookApi.onDidCreateOutput(renderOutput);
 
+// Copy of vscode-notebook-renderer old types as of 1.48
+// Keep these so we can support both the old interface and the new interface
+// Interface change here: https://github.com/DefinitelyTyped/DefinitelyTyped/pull/51675/files
+interface OldNotebookCellOutputMetadata {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    custom?: { [key: string]: any };
+}
+interface OldNotebookOutput {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    data: { [mimeType: string]: any };
+    metadata?: OldNotebookCellOutputMetadata;
+}
+interface OldNotebookOutputEventParams {
+    element: HTMLElement;
+    outputId: string;
+    output: OldNotebookOutput;
+    mimeType: string;
+}
+
 /**
  * Called from renderer to render output.
  * This will be exposed as a public method on window for renderer to render output.
  */
 function renderOutput(request: NotebookOutputEventParams) {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const mimeString = request.mime || (request as any).mimeType;
     try {
-        console.error('request', request);
+        console.log('request', request);
         const output = convertVSCodeOutputToExecutResultOrDisplayData(request);
-        console.log(`Rendering mimeType ${request.mimeType}`, output);
-        console.error('request output', output);
+        console.log(`Rendering mimeType ${mimeString}`, output);
+        console.log('request output', output);
 
-        ReactDOM.render(React.createElement(CellOutput, { mimeType: request.mimeType, output }, null), request.element);
+        ReactDOM.render(React.createElement(CellOutput, { mimeType: mimeString, output }, null), request.element);
     } catch (ex) {
-        console.error(`Failed to render mime type ${request.mimeType}`, ex);
+        console.error(`Failed to render mime type ${mimeString}`, ex);
     }
 }
 
 function convertVSCodeOutputToExecutResultOrDisplayData(
     request: NotebookOutputEventParams
 ): nbformat.IExecuteResult | nbformat.IDisplayData {
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    const metadata: Record<string, any> = {};
-    // Send metadata only for the mimeType we are interested in.
-    const customMetadata = request.output.metadata?.custom;
-    if (customMetadata) {
-        // Support for Old API
-        if (customMetadata[request.mimeType]) {
-            metadata[request.mimeType] = customMetadata[request.mimeType];
-        }
-        if (customMetadata.needs_background) {
-            metadata.needs_background = customMetadata.needs_background;
-        }
-        if (customMetadata.unconfined) {
-            metadata.unconfined = customMetadata.unconfined;
-        }
+    if ('mime' in request) {
+        // New API
+        return {
+            data: {
+                [request.mime]: request.value
+            },
+            metadata: request.metadata || {},
```

Comment:
Metadata is the bit that I want to check in particular when I look at the old API with stable. The new layout is a bit odd. There is a top-level metadata and then another unlabeled metadata just below that. I think it's the old custom metadata as it has needs_background in it.

![image](https://user-images.githubusercontent.com/812783/111361558-0e913a80-864b-11eb-8643-5851f58480d5.png)

Reply:
Mostly the naming seems odd. metadata.metadata feels vague to me.

Reply:
Should we give VS Code that feedback?